File: database-driver-plugin/src/main/java/de/lino/database/database/sql/postgresql/PostgreSQLDatabaseProvider.java
package de.lino.database.database.sql.postgresql;

import de.lino.database.database.auth.Credentials;
import de.lino.database.database.DatabaseType;
import de.lino.database.database.sql.SQLDatabaseProvider;
import de.lino.database.database.sql.SQLExecution;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link SQLDatabaseProvider} for PostgreSQL, connected via {@link SQLExecution} using
 * {@link DatabaseType#POSTGRES_SQL}'s JDBC driver and URL scheme.
 */
@Getter
public class PostgreSQLDatabaseProvider extends SQLDatabaseProvider {

    /**
     * Connects to a PostgreSQL database with {@code credentials}.
     *
     * @param credentials the login credentials and connection details to connect with
     */
    public PostgreSQLDatabaseProvider(@NotNull Credentials credentials) {
        super(DatabaseType.POSTGRES_SQL, new SQLExecution(DatabaseType.POSTGRES_SQL, credentials));
    }
    
}

File: python/database-driver-plugin/src/database_driver/plugin/database/sql/postgresql/postgresql_database_provider.py
"""Mirror of ``de.lino.database.database.sql.postgresql.PostgreSQLDatabaseProvider``."""

from __future__ import annotations

from database_driver.api.database.auth.credentials import Credentials
from database_driver.api.database.database_type import DatabaseType

from database_driver.plugin.database.sql.sql_database_provider import SQLDatabaseProvider
from database_driver.plugin.database.sql.sql_execution import SQLExecution


class PostgreSQLDatabaseProvider(SQLDatabaseProvider):
    """The ``SQLDatabaseProvider`` for PostgreSQL, connected via ``SQLExecution`` using
    ``DatabaseType.POSTGRES_SQL``'s driver and connection scheme."""

    def __init__(self, credentials: Credentials) -> None:
        super().__init__(DatabaseType.POSTGRES_SQL, SQLExecution(DatabaseType.POSTGRES_SQL, credentials))
